bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/dto/ActionTypeDTOMapper.java

```diff
@@ -29,15 +31,28 @@ public static ActionTypeDTO map(final ActionType actionType) {
         return actionTypeDto;
     }
 
+    public static CompositeActionTypeDTO map(final CompositeActionType actionType) {
```

Comment:
If we ever add an attribute to the ActionType (same for condition and trigger), it is also added to the CompositeActionType because it is a derived class.
Shouldn't we use a function that handles the ActionType[DTO] mapping that could be called by the mapping of a CompositeActionTypeDTO, too?

e.g.

```Java
    private static <T extends ActionTypeDTO> T map(final ActionType actionType, final T actionTypeDto) {
        fillProperties(actionType, actionTypeDto);
        actionTypeDto.inputs = actionType.getInputs();
        actionTypeDto.outputs = actionType.getOutputs();
        return actionTypeDto;
    }

    public static ActionTypeDTO map(final ActionType actionType) {
        return map(actionType, new ActionTypeDTO());
    }

    public static CompositeActionTypeDTO map(final CompositeActionType actionType) {
        final CompositeActionTypeDTO actionTypeDto = map(actionType, new CompositeActionTypeDTO());
        actionTypeDto.children = ActionDTOMapper.map(actionType.getChildren());
        return actionTypeDto;
    }
```

WDYT?